java/src/jmri/jmrit/display/layoutEditor/TrackSegment.java

```diff
@@ -299,6 +299,17 @@ private String getConnectName(Object o, int type) {
         return result;
     }
 
+    // These two methods should never be called…
```

Comment:
TrackSegment is the only LayoutTrack subclass that doesn't use [get/set]Connection (the other five subclasses do). Its setNewConnect1() and setNewConnect2() methods are used instead.
If I don't implement something here then I get the  "does not override abstract method " compiler error.
Are you saying I should declare it private here?
Or should I change everywhere that calls setNewConnect[1/2]() to use this method instead?

Reply:
The setNewConnect[1/2]() methods are only used by the LayoutEditor when you're dragging to create a new TrackSegment… it's already been initialized… (when you clicked the mouse) and connected at one end and now you're dragging the (unconnected) end to connect it to something. If you release the mouse without connecting the TrackSegment it will be removed. The addTrackSegment() method should be protected (because it relies on private LayoutEditor variables (beginObject, beginPointType, foundObject, foundPointType, dashedLine & mainlineTrack) to work correctly).

Reply:
Since, in this PR, you decided these methods must be implemented (by making them abstract in the parent class), you *are* expecting them to be called against any LayoutTrack instance (even if that instance is a TrackSegment).

I'm asking that the comment be more like:
```
/**
 * This implementation returns null because a TrackSegment has multiple connections, and
 * {@link #getConnect1() } and {@link #getConnect2() } should be used instead.
 *
 * @return always null
 */
```
that is, explaining why it returns null, instead of noting that you have a requirement you placed on this class that you don't want to have.

Reply:
You may also want to have this method (and setConnection(...)) get/set Connect1, and document that there are additional getters/setters for the second connection, and that this method only returns the first connection.

Reply:
Done…